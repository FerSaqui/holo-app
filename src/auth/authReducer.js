/**
 * En un reducer, la acciÃ³n va a modificar el state y react va a renderiza/actualizar lo que tenga que actualizar
 */

import { types } from "../types/types";

export const authReducer = (state = {}, action) => {

    switch ( action.type ) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }

        case types.logout:
            return {
                logged: false
            }
    
        default:
            return state;
    }

}