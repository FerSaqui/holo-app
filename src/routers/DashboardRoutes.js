import React from 'react';
import { Route, Routes } from 'react-router-dom';

import { HoloScreenEN } from '../components/EN/HoloScreenEN';
import { HoloScreenJP } from '../components/JP/HoloScreenJP';
import { Navbar } from '../components/Navbar/Navbar'
import { SearchScreen } from '../components/Search/SearchScreen'
import { VtuberScreen } from '../components/Vtuber/VtuberScreen';

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar />

            <div className="container">
                <Routes>
                    <Route path="holoJP" element={<HoloScreenJP />} />
                    <Route path="holoEN" element={<HoloScreenEN />} />

                    <Route path="vtuber/:vtuberId" element={<VtuberScreen />} />
                    <Route path="search" element={<SearchScreen />} />

                    <Route path="/" element={<HoloScreenJP />} />
                </Routes>
            </div>

        </>
    )
}
