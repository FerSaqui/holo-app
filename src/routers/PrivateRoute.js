import React, { useContext } from 'react'
import { Navigate, useLocation } from 'react-router-dom';
import { AuthContext } from '../auth/autContext';

export const PrivateRoute = ({children}) => {
    const { user } = useContext(AuthContext);

    /**
     * Para guardar la ultima pagina antes de cerrar sesi√≥n
     */
    const { pathname, search } = useLocation();

    /**
     * Guardar en el localstorage la ultima pagina antes de cerrar sesion
     */
    localStorage.setItem("lastPage", pathname + search);

  return user.logged
    ? children
    : <Navigate to={"/login"} />
}
