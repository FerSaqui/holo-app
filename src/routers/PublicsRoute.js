import React, { useContext } from 'react'
import { Navigate } from 'react-router-dom';
import { AuthContext } from '../auth/autContext';

export const PublicsRoute = ({ children }) => {
    const { user } = useContext(AuthContext);

    return user.logged
        ? <Navigate to={"/holoJP"} />
        : children
}
