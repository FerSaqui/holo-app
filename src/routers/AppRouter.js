import React from 'react'
import { Route, Routes, BrowserRouter } from 'react-router-dom'

import { LoginScreen } from '../components/Login/LoginScreen'
import { DashboardRoutes } from './DashboardRoutes'

export const AppRouter = () => {
    return (
        <BrowserRouter>
            <Routes>
                <Route path="/login" element={<LoginScreen />} />

                {/**Si la ruta no es /login, las demás rutas serán manejadas acá */}
                <Route path="/*" element={ <DashboardRoutes /> }/>
            </Routes>
        </BrowserRouter>
    )
}
