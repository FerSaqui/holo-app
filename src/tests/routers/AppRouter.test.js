import { mount } from "enzyme";
import { AuthContext } from "../../auth/autContext";
import { AppRouter } from "../../routers/AppRouter";

describe('Pruebas en <AppRouter />', () => {
    
    test('Debe de mostrar el login si no esta autenticado', () => {
        const contextValue = {
            user: {
                logged: false
            }
        };

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("h1").text()).toBe("Login Screen");
    });

    test('Debe de mostrar el componente de HoloJP si esta autenticado', () => {
        const contextValue = {
            user: {
                logged: true,
                name: "Fernando"
            }
        };
        
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find(".navbar").exists() ).toBe( true );
    });
});