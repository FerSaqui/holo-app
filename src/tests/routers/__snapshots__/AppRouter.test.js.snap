// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> Debe de mostrar el componente de HoloJP si esta autenticado 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Hololive
      </a>
      <button
        aria-controls="navbarText"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarText"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarText"
      >
        <ul
          className="navbar-nav me-auto mb-2 mb-lg-0"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/holoEN"
              onClick={[Function]}
            >
              Holo EN
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              href="/holoJP"
              onClick={[Function]}
            >
              Holo JP
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/search"
              onClick={[Function]}
            >
              Search
            </a>
          </li>
        </ul>
        <ul
          className="navbar-nav"
        >
          <span
            className="nav-item text-white ms-5"
            id="txtNombre"
          >
            Fernando
          </span>
          <button
            className="nav-item bg-danger"
            onClick={[Function]}
          >
            Logout
          </button>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div
      className="container"
    >
      <h1>
        HoloScreen JP
      </h1>
      <div
        className="row animate__animated animate__fadeIn"
      >
        <div
          className="col-12 col-md-6 col-lg-4 my-5 text-center animate__animated animate__fadeIn"
        >
          <div
            className="card text-white"
          >
            <img
              alt="Hoshimachi Suisei"
              className="card-img img-fluid"
              src="/assets/JP-Hoshimachi-Suisei.png"
            />
            <div
              className="card-img-overlay"
            >
              <a
                href="/vtuber/JP-Hoshimachi-Suisei"
                onClick={[Function]}
              >
                <h5
                  className="card-title bg-info rounded-pill"
                >
                  Hoshimachi Suisei
                </h5>
              </a>
              <p
                className="card-text bg-warning rounded-pill"
              >
                Hololive JP
              </p>
              <p
                className="card-text bg-info rounded position-absolute bottom-0 start-0"
              >
                On 12 November, she debuted her third 2D costume, in this costume she wears a black dress with a blue jacket, both the dress and the jacket are decorated with blue ribbons and laces, she wears a black belt with the words: 'Sphere' and 'Comet', she uses orange socks and black and short black boots
              </p>
            </div>
          </div>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 my-5 text-center animate__animated animate__fadeIn"
        >
          <div
            className="card text-white"
          >
            <img
              alt="Minato Aqua"
              className="card-img img-fluid"
              src="/assets/JP-Minato-Aqua.png"
            />
            <div
              className="card-img-overlay"
            >
              <a
                href="/vtuber/JP-Minato-Aqua"
                onClick={[Function]}
              >
                <h5
                  className="card-title bg-info rounded-pill"
                >
                  Minato Aqua
                </h5>
              </a>
              <p
                className="card-text bg-warning rounded-pill"
              >
                Hololive JP
              </p>
              <p
                className="card-text bg-info rounded position-absolute bottom-0 start-0"
              >
                On 26 September, Aqua debuted with a new 2D Live costume. She now wears two large braids, with the two small braids on each side of her bangs from her original design, decorated with a light navy blue headband. Her white top resembles a sailor fuku, with a pastel navy blue ribbon bow, and a row of little buttons going down the center.
              </p>
            </div>
          </div>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 my-5 text-center animate__animated animate__fadeIn"
        >
          <div
            className="card text-white"
          >
            <img
              alt="Momosuzu Nene"
              className="card-img img-fluid"
              src="/assets/JP-Momosuzu-Nene.png"
            />
            <div
              className="card-img-overlay"
            >
              <a
                href="/vtuber/JP-Momosuzu-Nene"
                onClick={[Function]}
              >
                <h5
                  className="card-title bg-info rounded-pill"
                >
                  Momosuzu Nene
                </h5>
              </a>
              <p
                className="card-text bg-warning rounded-pill"
              >
                Hololive JP
              </p>
              <p
                className="card-text bg-info rounded position-absolute bottom-0 start-0"
              >
                Nene's debut stream took place on 13 August 2020. She reached 100,000 subscribers that same day. Nene was the second member of hololive 5th generation to audition.
              </p>
            </div>
          </div>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 my-5 text-center animate__animated animate__fadeIn"
        >
          <div
            className="card text-white"
          >
            <img
              alt="Omaru Polka"
              className="card-img img-fluid"
              src="/assets/JP-Omaru-Polka.png"
            />
            <div
              className="card-img-overlay"
            >
              <a
                href="/vtuber/JP-Omaru-Polka"
                onClick={[Function]}
              >
                <h5
                  className="card-title bg-info rounded-pill"
                >
                  Omaru Polka
                </h5>
              </a>
              <p
                className="card-text bg-warning rounded-pill"
              >
                Hololive JP
              </p>
              <p
                className="card-text bg-info rounded position-absolute bottom-0 start-0"
              >
                She was introduced on hololiveâ€™s YouTube channel in a small introductory video along with the rest of the fifth gen on 6 August 2020.
              </p>
            </div>
          </div>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 my-5 text-center animate__animated animate__fadeIn"
        >
          <div
            className="card text-white"
          >
            <img
              alt="Sakura Miko"
              className="card-img img-fluid"
              src="/assets/JP-Sakura-Miko.png"
            />
            <div
              className="card-img-overlay"
            >
              <a
                href="/vtuber/JP-Sakura-Miko"
                onClick={[Function]}
              >
                <h5
                  className="card-title bg-info rounded-pill"
                >
                  Sakura Miko
                </h5>
              </a>
              <p
                className="card-text bg-warning rounded-pill"
              >
                Hololive JP
              </p>
              <p
                className="card-text bg-info rounded position-absolute bottom-0 start-0"
              >
                On 24 January, she participated in hololive 1st fes. Nonstop Story, along with the other hololive girls up to the third generation, where she debuted her 3D idol outfit.
              </p>
            </div>
          </div>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 my-5 text-center animate__animated animate__fadeIn"
        >
          <div
            className="card text-white"
          >
            <img
              alt="Shirakami Fubuki"
              className="card-img img-fluid"
              src="/assets/JP-Shirakami-Fubuki.png"
            />
            <div
              className="card-img-overlay"
            >
              <a
                href="/vtuber/JP-Shirakami-Fubuki"
                onClick={[Function]}
              >
                <h5
                  className="card-title bg-info rounded-pill"
                >
                  Shirakami Fubuki
                </h5>
              </a>
              <p
                className="card-text bg-warning rounded-pill"
              >
                Hololive JP
              </p>
              <p
                className="card-text bg-info rounded position-absolute bottom-0 start-0"
              >
                On 17 October, she debuted her first 3D model becoming the fourth hololive member to do so after Tokino Sora, Roboco, and Sakura Miko.
              </p>
            </div>
          </div>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 my-5 text-center animate__animated animate__fadeIn"
        >
          <div
            className="card text-white"
          >
            <img
              alt="Tokoyami Towa"
              className="card-img img-fluid"
              src="/assets/JP-Tokoyami-Towa.png"
            />
            <div
              className="card-img-overlay"
            >
              <a
                href="/vtuber/JP-Tokoyami-Towa"
                onClick={[Function]}
              >
                <h5
                  className="card-title bg-info rounded-pill"
                >
                  Tokoyami Towa
                </h5>
              </a>
              <p
                className="card-text bg-warning rounded-pill"
              >
                Hololive JP
              </p>
              <p
                className="card-text bg-info rounded position-absolute bottom-0 start-0"
              >
                On 1 January, she debuted her second 2D costume, a new year furisode during a stream, this costume has two different hairstyles one with a side ponytail and another with short hair, the flowers ornament can be used with either hairstyle.
              </p>
            </div>
          </div>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 my-5 text-center animate__animated animate__fadeIn"
        >
          <div
            className="card text-white"
          >
            <img
              alt="Tsunomaki Watame"
              className="card-img img-fluid"
              src="/assets/JP-Tsunomaki-Watame.png"
            />
            <div
              className="card-img-overlay"
            >
              <a
                href="/vtuber/JP-Tsunomaki-Watame"
                onClick={[Function]}
              >
                <h5
                  className="card-title bg-info rounded-pill"
                >
                  Tsunomaki Watame
                </h5>
              </a>
              <p
                className="card-text bg-warning rounded-pill"
              >
                Hololive JP
              </p>
              <p
                className="card-text bg-info rounded position-absolute bottom-0 start-0"
              >
                Watame was introduced along with the rest of the fourth gen on 25 December, 2019. Her Twitter account was created on November 2019, while her first tweet was made on 25 December, her YouTube channel was created on 4 December and her debut stream was on 29 December.
              </p>
            </div>
          </div>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 my-5 text-center animate__animated animate__fadeIn"
        >
          <div
            className="card text-white"
          >
            <img
              alt="Yukihana Lamy"
              className="card-img img-fluid"
              src="/assets/JP-Yukihana-Lamy.png"
            />
            <div
              className="card-img-overlay"
            >
              <a
                href="/vtuber/JP-Yukihana-Lamy"
                onClick={[Function]}
              >
                <h5
                  className="card-title bg-info rounded-pill"
                >
                  Yukihana Lamy
                </h5>
              </a>
              <p
                className="card-text bg-warning rounded-pill"
              >
                Hololive JP
              </p>
              <p
                className="card-text bg-info rounded position-absolute bottom-0 start-0"
              >
                She was introduced on hololive's YouTube channel in a small introductory video along with the rest of the fifth gen on 6 August.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Pruebas en <AppRouter /> Debe de mostrar el login si no esta autenticado 1`] = `
<div
  className="container my-5"
>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
